src/peer/negotiator.js

```diff
@@ -477,6 +477,52 @@ class Negotiator extends EventEmitter {
   }
 
   /**
+   * Get map object describes which kinds of tracks should be marked as recvonly
+   * @param {Object} options - Options of peer.call()
+   * @return {Object} Map object which streamTrack will be recvonly or not
+   */
+  _getReceiveOnlyState(options) {
+    const state = {
+      audio: false,
+      video: false,
+    };
+
+    const stream = options.stream;
+    const hasStream = stream instanceof MediaStream;
+    const hasAudioTrack = hasStream ? stream.getAudioTracks().length !== 0 : false;
+    const hasVideoTrack = hasStream ? stream.getVideoTracks().length !== 0 : false;
+
+    // force true if stream not passed(backward compatibility)
+    if (
+      hasStream === false
+      && options.audioReceiveEnabled === undefined
+      && options.videoReceiveEnabled === undefined
+    ) {
+      state.audio = true;
+      state.video = true;
+      return state;
+    }
+
+    // Set recvonly to true if `stream does not have track` and `option is true` case only
+    if (options.audioReceiveEnabled && hasAudioTrack === false) {
+      state.audio = true;
+    }
+    if (options.videoReceiveEnabled && hasVideoTrack === false) {
+      state.video = true;
+    }
+
+    // If stream has track, ignore options
```

Comment:
Could add the comment that expresses `having stream track leads to sendrecv` like below?

```
// If stream has track, ignore options, which results in setting sendrecv internally.
```

The reason of my comment is just for code readability and wants to stimulate my memory when I'll read the code in the future.